src/momento/utilities/expiration.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+import math
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import Optional
+
+
+class Expiration(ABC):
+    """Represents an expiration time for a token."""
+
+    _does_expire: bool
+
+    def __init__(self, does_expire: bool) -> None:
+        self._does_expire = does_expire
```

Comment:
Same situation in this file: `_does_expire` declared like this is a class variable, but you want an instance variable. You should omit it.